crates/rpc/src/traits.rs

```diff
@@ -0,0 +1,72 @@
+use core::time::Duration;
+
+use crate::api::SupervisorApiClient;
+use alloy_primitives::Log;
+use alloy_sol_types::SolEvent;
+use maili_interop::ExecutingMessage as InteropMessage;
+use maili_interop::CROSS_L2_INBOX_ADDRESS;
+use maili_protocol::ExecutingMessage as ProtocolMessage;
```

Comment:
@emhane I have the two `ExecutingMessage` types  here just for the sake of compilation atm.

The snippet you shared for parsing logs for `ExecutingMessage` relies on the interop crate. But the `SupervisorApiClient` uses the struct from protocol crate.

WDYT is the correct path here? E.G. I could use the protocol crate version and try to use `ExecutingMessage::abi_decode` instead. Or I could update the `SupervisorApiClient` to use the interop crate `ExecutingMessage`.

There are also some misc TODOs if you have time to weigh in on would be appreciated.

TY!

Reply:
update the `SupervisorApiClient` to use the interop crate `ExecutingMessage` - thank you !

Reply:
Thanks @emhane the logic is in place now. How do you recommend I proceed with unit testing this?

I can see in the repo you guys have implemented manual mocks elsewhere. In this instance I would have to mock the SupervisorApiClient and perhaps a struct that impls the ExecutingMessageValidator trait. In the past I have used mockall. LMK any suggestions